frontend/src/components/nodes/Proposals.tsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+
+import NodesApi, * as N from "../../libraries/explorer-wamp/nodes";
+
+import ProposalRow from "./ProposalRow";
+import PaginationSpinner from "../utils/PaginationSpinner";
+
+interface State {
+  proposals?: N.Proposal[];
+}
+
+export default class extends React.Component<State> {
+  state: State = {};
+
+  getNodes = async () => {
+    let proposals = (await new NodesApi().queryNodeRpc()).current_proposals;
+    this.setState({ proposals });
+  };
+
+  componentDidMount() {
+    this.getNodes();
+  }
+
+  render() {
+    const { proposals } = this.state;
+    let nodes;
+    if (proposals) {
+      nodes = proposals.map((node: N.Proposal) => (
+        <ProposalRow key={node.account_id} node={node} />
+      ));
+    }
+    if (!nodes) {
+      return <PaginationSpinner hidden={false} />;
+    }
+    return <>{nodes}</>;
```

Comment:
This can be simplified to (please, update the other components accordingly):

```suggestion
    if (!proposals) {
      return <PaginationSpinner hidden={false} />;
    }
    return <>
      {proposals.map((node: N.Proposal) => (
        <ProposalRow key={node.account_id} node={node} />
      ))}
    </>;
```